app/models/wakes/location.rb

```diff
@@ -3,17 +3,24 @@ class Wakes::Location < ActiveRecord::Base
   include Wakes::Metrics::GoogleAnalyticsPageviews
   include Wakes::Metrics::Facebook
 
-  validates :path, :format => { :with => %r{\A\/} }, :uniqueness => true
+  validates :path, :format => { :with => %r{\A\/} }, :uniqueness => { :scope => :host }
   belongs_to :resource, :foreign_key => :wakes_resource_id, :inverse_of => :locations, :touch => true
 
   scope :canonical, -> { where(:canonical => true) }
   scope :legacy, -> { where(:canonical => false) }
 
   def label
-    path
+    (host + path if host.present?) ||
+      path
   end
 
-  def url(protocol: 'http', host: ENV['DEFAULT_HOST'])
-    "#{protocol}://#{host}#{path}"
+  def url(protocol: 'http', host_override: nil)
+    calculated_hostname = host_override || host || ENV['DEFAULT_HOST']
+    "#{protocol}://#{calculated_hostname}#{path}"
+  end
+
+  def path_or_url
+    (path if host.blank?) ||
```

Comment:
if you want to be this concise, use a ternary operator: `host.blank? ? path : url`

I would probably flip it around to check presence, because that seems to read more logically to me: `host.present? ? url : path`